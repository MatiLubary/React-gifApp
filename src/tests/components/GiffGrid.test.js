import React from 'react'
import { shallow } from 'enzyme';
import {GifGrid} from "../../components/GifGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')


describe('Pruebas en <GifGrid />', () => {
    
    const category = "Dragonball";

    test('should display <GifGrid /> correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
            
        const wrapper = shallow( <GifGrid category={category} /> )
        expect(wrapper).toMatchSnapshot()

    });


    test('should display items when images are loaded', () => {

        const gifs = [{
            id: "abc",
            url: "https://localhost/wharever/cosa.jpg",
            title: "whareever"
        }]

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
            
        const wrapper = shallow( <GifGrid category={category} /> )

        expect(wrapper).toMatchSnapshot


   
    });

})