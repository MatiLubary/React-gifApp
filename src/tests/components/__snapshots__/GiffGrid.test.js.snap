// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> should display <GifGrid /> correctly 1`] = `
<div>
  <h3>
    Dragonball
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</div>
`;
